library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/MiscUtils.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.utils
+
+import android.app.Activity
+import android.content.res.Resources.Theme
+import android.graphics.Point
+import android.os.Build
+import android.util.TypedValue
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
+import com.datadog.android.sessionreplay.internal.recorder.ViewGlobalBounds
+import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
+
+internal object MiscUtils {
+
+    fun resolveThemeColor(theme: Theme): Int? {
+        val a = TypedValue()
+        theme.resolveAttribute(android.R.attr.windowBackground, a, true)
+        return if (a.type >= TypedValue.TYPE_FIRST_COLOR_INT &&
+            a.type <= TypedValue.TYPE_LAST_COLOR_INT
+        ) {
+            // windowBackground is a color
+            a.data
+        } else {
+            null
+        }
+    }
+
+    fun resolveSystemInformation(activity: Activity): SystemInformation {
+        return SystemInformation(
+            screenBounds = resolveScreenBounds(activity),
+            screenOrientation = activity.resources.configuration.orientation
+        )
+    }
+
+    @Suppress("DEPRECATION")
+    private fun resolveScreenBounds(activity: Activity): ViewGlobalBounds {
+        val windowManager = activity.windowManager ?: return ViewGlobalBounds(0, 0, 0, 0)
```

Comment:
in which case `windowManager` can be `null`?

Reply:
well not sure because I could not get this information from under the hood but I remember it happened to me at some point so I just wanted to make sure just in case. That method is not marked as `@NonNull` in Java.

Reply:
yeah...only if I would have that available in the code yet :). I have a big TODO task to add all the logs in the `session-replay` module. I will add a todo here too.